app/scripts/modules/google/src/interceptors/iap.interceptor.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  IHttpInterceptor,
+  IHttpPromiseCallbackArg,
+  IHttpProvider,
+  IPromise,
+  IQService,
+  IRequestConfig,
+  module,
+} from 'angular';
+import { $http } from 'ngimport';
+import { SETTINGS } from '@spinnaker/core';
+import { Dictionary } from 'lodash';
+
+/**
+ * When a Cloud IAP (if enabled) session expires, refreshes session.
+ */
+
+export class IapInterceptor implements IHttpInterceptor {
+  private MAX_RETRIES = 4;
+  private retryQueue: Dictionary<number> = {};
+
+  constructor(private $q: IQService) {
```

Comment:
Doesn't really matter, but this can be imported from `'ngimport'` as well. If you go that route, remove this constructor.

Reply:
Done.